### PR TITLE
Add environment variables for agent mode and download key

### PR DESCRIPTION
**Pull request by dusansusic**

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -8,6 +8,10 @@ zone:
 agent:
   # agent.key is the secret token which your agent uses to authenticate to Instana's servers.
   key: null
+  # agent.mode is used to set agent mode and it can be APM, INFRASTRUCTURE, or AWS
+  # mode: APM
```
Comment (dlbock) [inline values.yaml]: @dusansusic Do we need to document this configuration on the `README` as well?

**Comment by dlbock** [comment]
Otherwise LGTM. I'll have to update the official helm chart PR as well, with this change: https://github.com/helm/charts/pull/12799

File: values.yaml (hunk 1 above)
Comment (dusansusic) [reply values.yaml]: Done.

**Comment by dlbock** [comment]
@dusansusic Added one line for `agent.mode` in the README config table: https://github.com/instana/instana-agent-helm-chart/pull/25/commits/893fc41d372588555c1b63756f084e1af3f2f221

**Pull request merged**